src/TCP/TCPClient.cpp

```diff
@@ -0,0 +1,112 @@
+/// \file      TCPClient.h
+/// \brief     Based on GitHub Project Asynchronous Sockets for C++ (https://github.com/eminfedar/async-sockets-cpp)
+/// \date      June 23, 2021
+/// \author    My Linh Wuerzburger
+
+#include "TCPClient.h"
+#include <cstring>
+
+TCPSocket::TCPSocket(std::function<void(int, std::string)> onError, int socket_id) : BaseSocket(onError, TCP, socket_id) {
+}
+
+int TCPSocket::send_message(const std::string& message) {
+    return this->send_message(message.c_str(), message.length());
+}
+
+int TCPSocket::send_message(const char *bytes, size_t bytes_length) {
+    if (this->is_closed)
+        return -1;
+
+    ssize_t sent;
+    if ((sent = send(this->sock, bytes, bytes_length, 0)) < 0) {
+        perror("send");
```

Comment:
`perror` is in `stdio.h` defined bu not included. Why not use `spdlogger`?

Reply:
daf3839